### PR TITLE
Syslogformat

### PR DESCRIPTION
**Pull request by Jayanthi-moorthi**

### CODE REVIEW

File: logging/alternateServiceLogger.go [hunk 1]
```diff
@@ -52,16 +54,46 @@ func NewZapSugaredServiceLogger(lc AltServiceLoggerConfig) AltServiceLogger {
...
+				CallerKey:      "caller",
+				MessageKey:     "msg",
+				StacktraceKey:  "stacktrace",
+				EncodeLevel:    zapcore.LowercaseLevelEncoder,
+				EncodeTime:     zapcore.EpochTimeEncoder,
+				EncodeDuration: zapcore.SecondsDurationEncoder,
+				//EncodeCaller:   zapcore.ShortCallerEncoder,
+			},
+
+			Facility:  LOG_LOCAL0,
```
Comment (rosasck) [inline logging/alternateServiceLogger.go]: what does this do?

File: logging/alternateServiceLogger.go [hunk 2]
```diff
@@ -52,16 +54,46 @@ func NewZapSugaredServiceLogger(lc AltServiceLoggerConfig) AltServiceLogger {
...
+			Hostname:  "localhost",
```
Comment (rosasck) [inline logging/alternateServiceLogger.go]: should this be an arg, incase its prod vs staging?

File: logging/servicelogger.go [hunk 3]
```diff
@@ -3,176 +3,276 @@ package logging
...
+				MessageKey:     "msg",
+				StacktraceKey:  "stacktrace",
+				EncodeLevel:    zapcore.LowercaseLevelEncoder,
+				EncodeTime:     zapcore.EpochTimeEncoder,
+				EncodeDuration: zapcore.SecondsDurationEncoder,
+				//EncodeCaller:   zapcore.ShortCallerEncoder,
+			},
+
+			Facility:  LOG_LOCAL0,
+			Hostname:  "localhost",
```
Comment (rosasck) [inline logging/servicelogger.go]: same as above

File: logging/servicelogger.go [hunk 4]
```diff
@@ -3,176 +3,276 @@ package logging
...
+		enc = NewSyslogEncoder(SyslogEncoderConfig{
+			EncoderConfig: zapcore.EncoderConfig{
+				NameKey:        "logger",
+				CallerKey:      "caller",
+				MessageKey:     "msg",
+				StacktraceKey:  "stacktrace",
+				EncodeLevel:    zapcore.LowercaseLevelEncoder,
+				EncodeTime:     zapcore.EpochTimeEncoder,
+				EncodeDuration: zapcore.SecondsDurationEncoder,
+				//EncodeCaller:   zapcore.ShortCallerEncoder,
```
Comment (rosasck) [inline logging/servicelogger.go:54]: what does this do ?

File: logging/syslogFormat.go [hunk 5]
```diff
@@ -0,0 +1,367 @@
+package logging
+
```
Comment (rosasck) [inline logging/syslogFormat.go]: does this require a certain version of go?

File: logging/syslogFormat.go [hunk 6]
```diff
@@ -0,0 +1,367 @@
...
+const (
+	version         = 1
+	severityMask    = 0x07
+	facilityMask    = 0xf8
+	nilValue        = "-"
+	timestampFormat = "2006-01-02T15:04:05-07:00"
+	maxHostnameLen  = 255
+	maxAppNameLen   = 48
+
+	NonTransparentFraming Framing = iota
```
Comment (rosasck) [inline logging/syslogFormat.go]: what does iota mean? SORRY for all the questions this is amazing

File: logging/syslogFormat.go (hunk 5 above)
Comment (Jayanthi-moorthi) [reply logging/syslogFormat.go]: No Katie. Its just the package where the new file "syslogformat" added.

**Review by rosasck — APPROVED** [review]
looks amazing would love to learn how you did this! I have lots to learn from youu

File: logging/syslogFormat.go (hunk 6 above)
Comment (Jayanthi-moorthi) [reply logging/syslogFormat.go]: Iota in Go is used to represent constant increasing sequences

File: logging/alternateServiceLogger.go (hunk 1 above)
Comment (Jayanthi-moorthi) [reply logging/alternateServiceLogger.go]: Facility (Numeric Code) used by the Syslog system to calculate the priority. We need to pass the facility and severity to generate Priority for the specific logs. I have added the list of Facility and Severity.

File: logging/alternateServiceLogger.go (hunk 2 above)
Comment (Jayanthi-moorthi) [reply logging/alternateServiceLogger.go]: Yes katie. I will update this to be get from Env file.

File: logging/servicelogger.go (hunk 3 above)
Comment (Jayanthi-moorthi) [reply logging/servicelogger.go]: Sure Katie. I will update now

File: logging/servicelogger.go (hunk 4 above)
Comment (Jayanthi-moorthi) [reply logging/servicelogger.go:54]: These are some in built functionalities of zapcore to customise the logging format. LowerCaseLevelEncoder - to display logging level(INFO,DEBUG,..) in lowercase. EpochTimeEncoder - time serialization.

File: logging/syslogFormat.go (hunk 6 above)
Comment (Jayanthi-moorthi) [reply logging/syslogFormat.go]: Iota in Go is used to represent constant increasing sequences

**Pull request merged**